resources/documentation/stories.rst

```diff
@@ -0,0 +1,467 @@
+
+Adding a new range of nodes
+---------------------------
+
+You may need to add a new range of nodes. You can do it manually, adding them
+one by one, or simply use a small bash script to generate the content you need.
+
+We assume here you need to generate a new range of c[1-4] of nodes, with ip
+on range 10.10.3.[1-4]. Same kind of ranges for their BMC and interconnect.
+
+Create a file /root/gen.sh with the following content:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  cat <<EOF > computes.yml
+  mg_computes:
+    children:
+      equipment_typeC:
+        hosts:
+  EOF
+  for ((i=1;i<=$1;i++)); do
+  cat <<EOF >> computes.yml
+          c$i:
+            bmc:
+              name: bc$i
+              ip4: 10.10.103.$i
+              mac:
+              network: ice1-1
+            network_interfaces:
+              - interface: enp0s9
+                ip4: 10.10.3.$i
+                mac:
+                network: ice1-1
+              - interface: ib0
+                ip4: 10.20.3.$i
+                network: interconnect-1
+  EOF
+  done
+
+Save, make this script executable, and launch it asking for 4 nodes:
+
+.. code-block:: text
+
+  chmod +x /root/gen.sh
+  /root/gen.sh 4
+
+You should now have a file named *computes.yml* inside your current folder with
+the desired content. Refer to single node add story seen above on how now update
+the cluster configuration.
+
+Adding a new master group
+-------------------------
+
+You may need to create a new master group, for a new kind of range of equipment.
+
+The stack is fully dynamic regarding groups. The only thing you need is to
+create a new file with the master group name inside of
+/etc/bluebanquise/inventory/cluster/nodes/
+
+For example, if you wish to create a new group "switchs", create file
+/etc/bluebanquise/inventory/cluster/nodes/switches.yml and add the following
+content in the file:
+
+.. code-block:: yaml
+
+  mg_switchs:
+    children:
+
+The master group is now created.
+
+Note that master groups must always be prefixed by the string *mg_* to be
+detected by the stack. It is also possible for advanced users to change this
+prefix pattern in the general_settings part.
+
+Adding a new equipment_profile group
+------------------------------------
+
+To create a new equipment profile, create its associated folder. We will assume
+here that you wish to create equipment profile equipment_X:
+
+.. code-block:: text
+
+  mkdir /etc/bluebanquise/inventory/group_vars/equipment_X
+
+Then, if this equipment need to be different than the generic equipment_profile
+configuration (/etc/bluebanquise/inventory/group_vars/all/equipment_all/),
+create new files into /etc/bluebanquise/inventory/group_vars/equipment_X and use
+Ansible precedence mechanism to set your settings.
+
+You can refer to the example inventories in resources/examples/ to see more of
+these files.
+
+You can now add nodes into this equipment profile. See adding nodes stories
+above.
```

Comment:
Can we create anchors to refer other user stories with rst?

Reply:
Sure ! I will do that in a next commit :-)

Reply:
Done